Models/MuZero_torch_agent.py

```diff
@@ -62,24 +64,22 @@ def __init__(self):
         super().__init__()
         self.full_support_size = config.ENCODER_NUM_STEPS
 
-        self.representation_network = mlp(config.OBSERVATION_SIZE, [config.HEAD_HIDDEN_SIZE] *
-                                          config.N_HEAD_HIDDEN_LAYERS, config.LAYER_HIDDEN_SIZE)
+        self.representation_network = mlp(config.OBSERVATION_SIZE, [config.LAYER_HIDDEN_SIZE] *
+                                          config.N_HEAD_HIDDEN_LAYERS, config.HIDDEN_STATE_SIZE)
 
-        self.action_encodings = mlp(config.ACTION_CONCAT_SIZE, [config.HEAD_HIDDEN_SIZE] * config.N_HEAD_HIDDEN_LAYERS,
-                                    config.LAYER_HIDDEN_SIZE)
+        self.action_encodings = mlp(config.ACTION_CONCAT_SIZE, [config.LAYER_HIDDEN_SIZE] * 0,
+                                    config.HIDDEN_STATE_SIZE)
+        
+        self.dynamics_hidden_state_network = torch.nn.LSTM(input_size = config.HIDDEN_STATE_SIZE, 
+                          num_layers = config.NUM_RNN_CELLS, hidden_size = config.LSTM_SIZE, batch_first = True).cuda()
 
-        self.dynamics_encoded_state_network = [
-            torch.nn.LSTMCell(config.LAYER_HIDDEN_SIZE, 256).cuda(),
-            torch.nn.LSTMCell(256, 256).cuda()
-        ]
+        self.dynamics_reward_network = mlp(config.HIDDEN_STATE_SIZE, [1] *
```

Comment:
why does this reward network have a layer of size 1

Reply:
I wanted to minimize the weights of the reward layer since I'm not using it. If this ends up working, I will change back to the default sizing like the others

Reply:
Probably change back so we can experiment with multiple rewards. Also, the tree still uses this reward output even if the trainer does not